### ISSUE TITLE
global-require complains when I believe it shouldn't

### ISSUE THREAD

**Issue by ronkorving** [eslint/eslint#4222]
ESLint 1.7.2 complains about this code:

```js
const foo = {
  bar: require('./bar')
};
```

The error: `Unexpected require()  global-require`

In my opinion, that should not raise alarm bells (it *is* at the top of the file). Looking at the already accepted cases under "The following patterns are not considered problems:" on [the rule page](http://eslint.org/docs/rules/global-require), I think this is not unreasonable, is it?

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
The rule is behaving as it was designed. We made the decision to not include anything other than `require` in variable declarations. You do have the option of creating a custom rule if you'd like some other behavior.

**Comment by ronkorving** [comment]
The thing is that the require call still executes before the creation of my object completes. Doesn't that fit the goal of this rule?

**Comment by platinumazure** [comment]
Seems to me like that's a can of worms. How many levels of nesting would
you want supported after that? Just put it in a variable and then use it in
your object. Not that hard :-)

**Comment by ronkorving** [comment]
Of course there are workarounds, but I have a case where I need to include a whole list of files (each exposing Express.js route handlers) and want to put them in an object, eg:

```js
const middleware = {
  foo: require('./middleware/foo'),
  bar: require('./middleware/bar'),
  etc: require('./middleware/etc')
};
```

I don't really wanna argue details about what I'm doing in my app, apart from making the point that this is a real use case and assigning the require() result to a variable first would dramatically clutter my code with noise, which is the opposite of what a linter is supposed to achieve (it's supposed to promote clarity, right?).
